src/screens/AccountPin.js

```diff
@@ -111,14 +117,14 @@ class AccountPinView extends React.PureComponent {
           onSubmitEditing={() => {
             this.setState({ focusConfirmation: true });
           }}
-          onChangeText={pin => this.setState({ pin: pin, pinMismatch: false, pinTooShort: false })}
+          onChangeText={this.onPinInputChange('pin')}
           value={this.state.pin}
         />
         <Text style={styles.title}>CONFIRM PIN</Text>
         <PinInput
           returnKeyType="done"
           focus={this.state.focusConfirmation}
-          onChangeText={confirmation => this.setState({ confirmation: confirmation, pinMismatch: false, pinTooShort: false })}
+          onChangeText={this.onPinInputChange('confirmation')}
```

Comment:
agreed with @Tbaut above, why not have a separate handler for pin and confirmation rather than trying to make it general within `onPinInputChange` ?